doc/troubleshooting.md

```diff
@@ -0,0 +1,31 @@
+[<- Back to Index](index.md)
+
+# Troubleshooting FAQ
+
+### I am unable to connect to the LRS when opening the UI
+
+First of all, ensure that the SQL LRS app is running and that the host and port are configured correctly. If you are running SQL LRS as a Docker image, ensure that the port is exposed.
+
+In addition, you may need to adjust [configuration](env_vars.md#cors) for CORS (Cross-Origin Resource Sharing), which restricts which endpoints requests can be sent from and be accepted by the LRS. Either specify allowed endpoint URLs via `LRSQL_ALLOWED_ORIGINS` (the recommended way for production instances) or allow all endpoints via setting `LRSQL_ALLOW_ALL_ORIGINS` to `true`.
```

Comment:
What do you mean "hosts file entries"? (And don't you mean "**host's** file entries"?)

Reply:
/etc/hosts. look up `hosts file`. It's a thing. Can't just write /etc/hosts because different OSs put it different places...

Reply:
Oh. They never taught us that part in Computer Networking (and why would they in a DNS-run world?)

Reply:
Regardless, I'm not sure if that specification is necessary, since during development I still get the same CORS-related errors if I send requests from localhost.

Reply:
do you? i thought localhost was an exception

Reply:
Ok turns out localhost _is_ an exception and I was just running FE on a different port from BE. So I added the extra language about domain names + proxies in https://github.com/yetanalytics/lrsql/pull/389/commits/de3feb712219160f4754ca9333d2c86511d8d0bc.

Reply:
BTW I can use the word "endpoints" in place of "origins," right, or is that incorrect? I did that in order to not have to say "origin" over and over again and give a sense of definitional redundancy.

Reply:
So looking at [the definition of "origin" on MDN](https://developer.mozilla.org/en-US/docs/Glossary/Origin), it seems like I could use "endpoint" and "origin" interchangeably here, since they both refer to protocol + hostname + port. It _would_ be inaccurate to use "URL" here though, since URLs would also include extra stuff like paths and query args.